### PR TITLE
Make recipes more themeable

### PR DESCRIPTION
**Pull request by SimonDanisch**

### CODE REVIEW

**Comment by SimonDanisch** [comment]
This adds more advanced theming capabilities to the model plots!
How and what to theme can be found in the docs: 
https://beacon-biosignals.github.io/Lighthouse.jl/previews/PR32/plotting/#Theming

This also makes sure, that text on barplots/heatmaps always has a high contrast. The disadvantage is, that one has less control over the color, since the contrast finding algorithm finds the lightness with the highest contrast, so often black/white come out of it, and the original text color is just used as guidance. This could be improved by having some bounds on the lightness.

I had to update the compat version to Julia 1.5. If I remember correctly, @hannahilea already was ok with such a bump for the last PR, so I hope that's not a show stopper!

I also added some more tests for some of our plotting corner cases (e.g. when the confusion matrix contains NaNs).

File: docs/src/plotting.md [hunk 1]
```diff
@@ -32,6 +32,32 @@ plot_confusion_matrix!(fig[1, 2], confusion, classes, :Column, annotation_text_s
 fig
 ```
 
+## Theming
+
+All plots are globally themeable, by setting there `camelcase(functionname)` to a theme. Usually, there are a few sub categories, for e.g. axis, text and subplots.
```
Comment (hannahilea) [inline docs/src/plotting.md:37]:
```suggestion
All plots are globally themeable, by setting their `camelcase(functionname)` to a theme. Usually, there are a few sub categories, for e.g. axis, text and subplots.
```

**Review by hannahilea — APPROVED** [review]
- [x] update incomplete docstring

File: docs/src/plotting.md [hunk 2]
```diff
@@ -56,6 +82,40 @@ Note that all curve plot types accepts these types:
 Lighthouse.XYVector
 Lighthouse.SeriesCurves
 ```
+## Theming
+All curves geth themed like this, with the main key (in this case `ReliabilityCalibrationCurves`), being the camel case variant of the function name and allowiing .
```
[the remark this answers is not in the document]
Comment (SimonDanisch) [reply docs/src/plotting.md:86]: Thought it was time for some suspense 😅

[the remark this answers is not in the document]
Comment (SimonDanisch) [reply docs/src/plotting.md]: Ok, lets see if that works better ;)
